packages/mui-base/src/useMenu/useMenu.ts

```diff
@@ -37,20 +37,28 @@ const FALLBACK_MENU_CONTEXT: DropdownContextValue = {
  * - [useMenu API](https://mui.com/base-ui/react-menu/hooks-api/#use-menu)
  */
 export function useMenu(parameters: UseMenuParameters = {}): UseMenuReturnValue {
-  const { listboxRef: listboxRefProp, onItemsChange, id: idParam } = parameters;
+  const {
+    listboxRef: listboxRefProp,
+    onItemsChange,
+    id: idParam,
+    disabledItemsFocusable = true,
+    open: openParameter,
```

Comment:
With this, I kind of made the hook to work without context too, I know we had a PR to make the prop required, but I don't see a better way in order to support both APIs. I am curious as to what are @michaldudak thoughts on this.

Reply:
In #39284 I'm also adding the `anchor` prop to the menu back, so this may suggest that the solution we chose in #32088 wasn't the optimal one.

Anyway, now the question here also is if we're adding this prop back only to support the existing Material UI-style API, or is there a real need for it? If the former, maybe we shouldn't have spent so much time on the new implementation of Base UI, and should have just extracted the unstyled bits from the Material UI component without changing the API. I've always seen Base UI as a fresh start without the strong requirement to have a 100% compatible API with Material UI.

---

All these aside, I'm wondering if it would be possible to use both the `useMenu` and `useDropdown` in the Material Next's Menu (so that the open/close state is handled by the dropdown, while the list navigation is handled by the menu).

Reply:
> All these aside, I'm wondering if it would be possible to use both the useMenu and useDropdown in the Material Next's Menu (so that the open/close state is handled by the dropdown, while the list navigation is handled by the menu).

I don't think this is the optimal solution, especially considering that most of the UI libraries would like to offer both controlled and non-controlled open state for the component, which means all of them would need to use both hooks - if this can be replaced with just an additional argument in the `useMenu` hook, why wouldn't we do it.

> Anyway, now the question here also is if we're adding this prop back only to support the existing Material UI-style API, or is there a real need for it? 

In my opinion, this would benefit Joy UI too. Considering this, it's likely going to benefit other libraries that would depend on Base UI.

> If the former, maybe we shouldn't have spent so much time on the new implementation of Base UI, and should have just extracted the unstyled bits from the Material UI component without changing the API. I've always seen Base UI as a fresh start without the strong requirement to have a 100% compatible API with Material UI.

Material UI is just one example of a UI library that already has an opinion on how things should work - this is why by using Base UI in it we discover things that can be improved. So far, Joy UI is adapting whatever Base UI has, which does not allow us to test a real-world use case most of the time. What I mean is, I don't think @siriwatknp would be opposed to these changes, but he won't necessarily push you to do them :D I am happy to try to refactor the component to use the `useDropdown` inside it too, but is that what we want to advertise users of the library to do? If yes, we should also add a demo about it on the Base UI's Menu page.

Reply:
I agree that we don't know from real-world usage which pattern would be the best. But also, being "the best" could mean something else to different people, so this can be hard to measure. Surely, one option is to provide more ways of doing things, but this comes with a price - more APIs to support and possibly confused users not knowing which API to choose (and perhaps some even more confused, who mix different APIs together). So I'd be in favor of having one API and teaching developers how to use it rather than having many and allowing everyone to choose. This doesn't mean it must be the existing API with a separate Dropdown and Menu - maybe it's Material UI that got it right.

For now let's try using both `useDropdown` and `useMenu` in Material UI's Menu to get a feeling about the DX and we can discuss it afterward.

Reply:
Here is the commit for using both hooks https://github.com/mui/material-ui/pull/38934/commits/1d6e37833c46b21f16f2e9fa7b18447232a07c66. The downside is that you need to create a wrapper component so that you can provide the dropdown context in the parent of the one that uses the `useMenu` hook.. Unless you had a different thought on how the two hooks can be combined. Seeing the changes in the Menu.js file, to me it is kind of ridiculous to ask people to do this, if we can easily support providing the `open` param. However, we can always add this as an additional API afterwards, so happy to try to see what will be the feedback.

Reply:
Thanks for trying it out!

> The downside is that you need to create a wrapper component so that you can provide the dropdown context in the parent of the one that uses the useMenu hook.

Yup, that's unfortunately necessary if you want to keep the existing API.

> Seeing the changes in the Menu.js file, to me it is kind of ridiculous to ask people to do this, if we can easily support providing the open param

Let's consider we expose the `open` parameter and not require the Dropdown to be present. Wouldn't this require developers to handle things like restoring focus on the menu button when a menu is closed or setting up `aria-controls` and `aria-labelledby` attributes on their own?

Reply:
> Wouldn't this require developers to handle things like restoring focus on the menu button when a menu is closed or setting up aria-controls and aria-labelledby attributes on their own?

You are right, this would not be a regression for Material UI, as it worked like that before, but it may be surprising for other people trying the hook... We can try to go as is for now and consider adding the API later if we see more people wanting to use the menu as a standalone component - I can't think of valid use-cases apart from the selfish backporting API for Material UI :) 

I will update the PR to make it closer to mergable state then without changing the Base UI API further then just adding the `disabledItemsFocusable` option that was missing.